index.js

```diff
@@ -1818,6 +1818,34 @@ Jimp.prototype.sepia = function (cb) {
 };
 
 /**
+ * Applies a fisheye lens to an image
+ * @param (optional) cb a callback for when complete
+ * @returns this for chaining of methods
+ */
+Jimp.prototype.fisheye = function (cb) {
+    var source = this.cloneQuiet();
+    var width = source.bitmap.width;
+    var height = source.bitmap.height;
+    var that = this;
+    source.scanQuiet(0, 0, width, height, function (x, y, idx) {
+        var hx = x / width;
+        var hy = y / height;
+        var r = Math.sqrt(Math.pow(hx - 0.5, 2) + Math.pow(hy - 0.5, 2));
+        var rn = 2 * (Math.pow(r, 2.5));
+        var cosA = (hx - 0.5)/r;
+        var sinA = (hy - 0.5)/r;
+        var newX = Math.round(((rn * cosA + 0.5) * width));
+        var newY = Math.round(((rn * sinA + 0.5) * height));
+
+        let color = source.getPixelColor(newX, newY);
+        that.setPixelColor(color, x, y);
+    });
+
```

Comment:
I would add `that.setPixelColor(source.getPixelColor(width / 2, height / 2), width / 2, height / 2);` here, because without it, it fails to set the center pixel to a color and makes it just transparent